### ISSUE TITLE
Go to definition of struct opens typedef from other file instead of immediate declaration

### ISSUE THREAD

**Issue by Zylann** [microsoft/vscode-cpptools#4135]
**Type: LanguageService**

Windows 10 64 bits
VSCode version: 1.37.1 (user setup)

I have two files in a freshly created folder:

alpaca.h
```cpp
template <typename T>
class Alpaca {
public:
	struct Test {
	};

	void foo(Test a) {
	}
};
```

cat.h
```cpp
#include "alpaca.h"

class Cat {
	typedef Alpaca<int> A;
	typedef A::Test Test;
};

```

When I open `alpaca.h`, and use `Go to definition` on the occurrence of `Test` in the `foo` function, it opens `cat.h` and shows me the typedefed `Test`. While this is indirectly correct, I would have expected intellisense to point me the direct declaration of `Test`, just above `foo`.

If there are more than one `typedef` defined in the same manner, it even opens a peek list of them, but still not the wanted one.

**Comment by sean-mcmanus** [comment]
I've verified the bug. It doesn't repro with VS.

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. It should be fixed in our 0.25.1 bug fix update for next week -- unless it gets "cut" for some reason.

**Comment by sean-mcmanus** [comment]
Fixed with 0.25.1.
